contracts/core/tokens/rebalancing-libraries/ProposeLibrary.sol

```diff
@@ -50,110 +52,91 @@ library ProposeLibrary {
     /* ============ Internal Functions ============ */
 
     /**
-     * Function used to validate inputs to propose function and initialize auctionParameters struct
+     * Function used to validate inputs to propose function
      *
-     * @param _nextSet                      The Set to rebalance into
-     * @param _auctionLibrary               The library used to calculate the Dutch Auction price
-     * @param _auctionTimeToPivot           The amount of time for the auction to go ffrom start to pivot price
-     * @param _auctionStartPrice            The price to start the auction at
-     * @param _auctionPivotPrice            The price at which the price curve switches from linear to exponential
-     * @param _componentWhiteListAddress    Component WhiteList address
-     * @param _proposeParameters            Rebalancing Set Token state parameters needed to execute logic
-     * @return                              Struct containing auction price curve parameters
+     * @param _nextSet                    The Set to rebalance into
+     * @param _auctionLibrary             The library used to calculate the Dutch Auction price
+     * @param _proposalContext            Rebalancing Set Token state parameters needed for proposal validation
      */
-    function propose(
+    function validateProposal(
         address _nextSet,
         address _auctionLibrary,
-        uint256 _auctionTimeToPivot,
-        uint256 _auctionStartPrice,
-        uint256 _auctionPivotPrice,
-        address _factoryAddress,
-        address _componentWhiteListAddress,
-        ProposeAuctionParameters memory _proposeParameters
+        ProposalContext memory _proposalContext,
```

Comment:
`Context`..? Prefer `AuctionParameters`

Reply:
Well.. the struct has the data that the function needs to be aware of and isn't auction-specific.